src/cr/cube/matrix/measure.py

```diff
@@ -824,3 +913,62 @@ def blocks(self):
         return SumDiffSubtotals.blocks(
             self._weighted_cube_counts.weighted_counts, self._dimensions
         )
+
+
+# === PAIRWISE HELPERS ===
+
+
+class _PairwiseSignificaneBetweenSubvariablesHelper(object):
+    """Helper for calculating overlaps significance between subvariables."""
+
+    def __init__(self, overlaps, valid_overlaps, idx_a, idx_b):
+        self._overlaps = overlaps
+        self._valid_overlaps = valid_overlaps
+        self._idx_a = idx_a
+        self._idx_b = idx_b
+
+    @lazyproperty
+    def p_vals(self):
+        """float64 significance p-vals for the selected subvariables."""
+        if self._idx_a == self._idx_b:
+            return 0.0
+
+        return 2 * (1 - t.cdf(abs(self.t_stats), df=self._N - 2))
+
+    @lazyproperty
+    def t_stats(self):
+        """float64 significance t-stats for the selected subvariables."""
+        if self._idx_a == self._idx_b:
+            return 0.0
+
+        Sa, Sb, Sab = self._selected_counts
+        Na, Nb, Nab = self._valid_counts
+        pa, pb, pab = Sa / Na, Sb / Nb, Sab / Nab
+
+        return (pa - pb) / np.sqrt(
+            1 / self._N * (pa * (1 - pa) + pb * (1 - pb) + 2 * pa * pb - 2 * pab)
+        )
+
+    @lazyproperty
+    def _N(self):
+        """int count of non-overlapping cases of subvariables a and b."""
+        Na, Nb, Nab = self._valid_counts
+        return Na + Nb - Nab
+
+    @lazyproperty
+    def _selected_counts(self):
+        """tuple(int) of selected counts for subvars a, b, and the overlaps (a^b)."""
+        return (
+            self._overlaps[self._idx_a, self._idx_a],
+            self._overlaps[self._idx_b, self._idx_b],
+            self._overlaps[self._idx_a, self._idx_b],
+        )
+
+    @lazyproperty
+    def _valid_counts(self):
+        """tuple(int) of valid counts for subvars a, b, and the overlaps (a^b)."""
+        return (
+            self._valid_overlaps[self._idx_a, self._idx_a],
+            self._valid_overlaps[self._idx_b, self._idx_b],
+            self._valid_overlaps[self._idx_a, self._idx_b],
+        )
```

Comment:
Well, now that you point it out... They _can_ actually be float, if a weight is used. But most of the times they won't be. So the answer to _why_ (when they're indeed ints) is that they come that way from ZZ9. Because it only adds together `True` or `False` values for each row of a subvariable (to figure out how many of those `row[idx] == 1` (when it's about selected overlaps) or to figure out `row[idx] != -1` (how many are valid)). Anyways, I'll change and elaborate.